sdk-base/src/main/java/com/mapbox/maps/plugin/Plugin.kt

```diff
@@ -1,42 +1,62 @@
 package com.mapbox.maps.plugin
 
 /**
- * Class name for compass plugin.
- */
-const val PLUGIN_COMPASS_CLASS_NAME = "com.mapbox.maps.plugin.compass.CompassViewPlugin"
-/**
- * Class name for logo plugin.
- */
-const val PLUGIN_LOGO_CLASS_NAME = "com.mapbox.maps.plugin.logo.LogoViewPlugin"
-/**
- * Class name for gestures plugin.
- */
-const val PLUGIN_GESTURE_CLASS_NAME = "com.mapbox.maps.plugin.gestures.GesturesPluginImpl"
-/**
- * Class name for map attribution plugin.
- */
-const val PLUGIN_ATTRIBUTION_CLASS_NAME = "com.mapbox.maps.plugin.attribution.AttributionViewPlugin"
-/**
- * Class name for location component plugin.
- */
-const val PLUGIN_LOCATION_COMPONENT_CLASS_NAME = "com.mapbox.maps.plugin.locationcomponent.LocationComponentPluginImpl"
-/**
- * Class name for scale bar plugin.
- */
-const val PLUGIN_SCALE_BAR_CLASS_NAME = "com.mapbox.maps.plugin.scalebar.ScaleBarPluginImpl"
-/**
- * Class name for camera animations plugin.
- */
-const val PLUGIN_CAMERA_ANIMATIONS_CLASS_NAME = "com.mapbox.maps.plugin.animation.CameraAnimationsPluginImpl"
-/**
- * Class name for map overlay plugin.
- */
-const val PLUGIN_MAPOVERLAY_CLASS_NAME = "com.mapbox.maps.plugin.overlay.MapOverlayPluginImpl"
-/**
- * Class name for annotation plugin.
- */
-const val PLUGIN_ANNOTATION_CLASS_NAME = "com.mapbox.maps.plugin.annotation.AnnotationPluginImpl"
-/**
- * Class name for lifecycle plugin.
- */
-const val PLUGIN_LIFE_CYCLE_CLASS_NAME = "com.mapbox.maps.plugin.lifecycle.MapboxLifecyclePluginImpl"
\ No newline at end of file
+ * Class describing plugin instance.
+ *
+ * For Mapbox plugins it is enough to provide plugin id from this class without plugin instance.
+ * For user-defined plugins both unique id and plugin instance must be provided.
+ */
+data class Plugin(
+  /**
+   * Unique plugin id.
+   */
+  val id: String,
+  /**
+   * Plugin instance implementing [MapPlugin].
+   * Must be non-null for user-defined plugins or RuntimeException will occur when trying to initiate such plugin.
+   */
+  val instance: MapPlugin? = null
```

Comment:
Since the instance parameter is only applicable for custom plugins other than Mapbox. Does it make sense to distinguish Mapbox Plugin and Custom Plugin, like a sealed class with Custom and Mapbox?  
```
sealed class Plugin(
  /**
   * Unique plugin id.
   */
  val id: String,
  /**
   * Plugin instance implementing [MapPlugin].
   * Must be non-null for user-defined plugins or RuntimeException will occur when trying to initiate such plugin.
   */
  val instance: MapPlugin? = null) {
    class Mapbox(id: String) : Plugin(id, null)
    class Custom(id: String, instance: MapPlugin)
  }
```